test/grpc_utils_test.py

```diff
@@ -145,6 +148,73 @@ async def wrapped_blob_create(req, **kwargs):
         await wrapped_blob_create.aio(req, timeout=4.0)
 
 
+def test_CustomProtoStatusDetailsCodec_round_trip():
+    blob_msg = api_pb2.BlobCreateResponse(blob_id="abc")
+    sandbox_msg = sandbox_router_pb2.SandboxExecPollResponse(code=31)
+    msgs = [blob_msg, sandbox_msg]
+
+    codec = CustomProtoStatusDetailsCodec()
+    encoded_msg = codec.encode(Status.CANCELLED, "this-is-a-message", msgs)
+    assert isinstance(encoded_msg, bytes)
+
+    decoded_status = api_pb2.RPCStatus.FromString(encoded_msg)
+    assert decoded_status.message == "this-is-a-message"
+    assert decoded_status.code == Status.CANCELLED.value
+
+    decoded_msg = codec.decode(Status.CANCELLED, None, encoded_msg)
+    assert len(decoded_msg) == 2
+    assert decoded_msg == msgs
+
+
+def test_CustomProtoStatusDetailsCodec_unknown():
+    encoded_details = [
+        Any(type_url="abc", value=b"bad"),
+    ]
+    encoded_msg = api_pb2.RPCStatus(details=encoded_details).SerializeToString()
+    codec = CustomProtoStatusDetailsCodec()
+
+    decoded_msg = codec.decode(Status.OK, None, encoded_msg)
```

Comment:
In normal grpclib usage is there any way to access details for an OK status? My understanding was they only are used to populate the `GRPCError` metadata (even though it's just more trailing metadata on the wire).